packages/engine/errors/1109.md

```diff
@@ -0,0 +1,20 @@
+---
+original: "Expression expected."
+excerpt: "I think there is a syntax error your code."
+---
+
+I don't like it when you declare a variable and export it in the same statement.
+
+```ts
+export default const foo: string = "this is a string"; 
+```
+
+I would prefer it if you export the variable on the next line instead.
+
+```ts
+const foo: string = "this is a string";
+export default foo;
+```
```

Comment:
This is not entirely accurate. An expression is (in short) something that resolves to a value, there are 5 categories of expressions in JavaScript (well really only two: with side-effects & no side-effects): [MDN: Expressions_and_Operators#expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_Operators#expressions)

For example, each of these 7 lines of code error with "Expression expected."
```ts
import {} from;
export default;
1 + ;
1 >=;
< 2;
if () {}
for () {}
```

So the description code is inaccurate:
```ts
export default const foo: string = "this is a string"; 
```
TS points at `const` because `const` is a keyword and not an expression. Since an expression is something that resolves a value, an assignment to a variable is also an expression (with side-effects), so this is perfectly valid:
```ts
let foo: string
export default foo = "this is a string";
```
A way to see that assignment is an expression with side-effects:
```ts
let foo: string
console.log(foo = "this is a string"); // this is a string
console.log(foo);                      // this is a string
```
It has side-effects because it changes the value of `foo` variable but at the same time it resolve with the value assigned to `foo`.

---

So an excerpt of something along the lines would be better:
> I was expecting a code that resolves to a value

As for the description, I believe this particular error can only be better described with examples, so you could use the above mentioned 7 examples to display what it actually means, sort of like:
```ts
import {} from; // I'm expecting an expression after 'from'
export default; // I'm expecting an expression after 'default'
1 + ;           // I'm expecting an expression after '+'
1 >=;           // I'm expecting an expression after '>='
< 2;            // I'm expecting an expression before '<'
if () {}        // I'm expecting an expression between '()'
for () {}       // I'm expecting an expression between '()'
```

Maybe it can be worded better with even simpler language.